pkg/controller/replicaset/replica_set.go

```diff
@@ -580,6 +580,15 @@ func (rsc *ReplicaSetController) syncReplicaSet(key string) error {
 		return err
 	}
 
+	//Do a full fetch here.
+	//The cache may be out of date via race condition #69376
+	fresh, err := rsc.kubeClient.AppsV1().ReplicaSets(rs.Namespace).Get(rs.Name, metav1.GetOptions{})
+	if err == nil && fresh.UID != rs.UID {
+		klog.Errorf("%v %v's UID in cache differed from server. This object was deleted.", rsc.Kind, key)
+		rsc.expectations.DeleteExpectations(key)
+		return nil
```

Comment:
Should this really return? Doesn't it need to proceed to process the new RS?

Reply:
The creation of the new RS shows up as a second event, and will get processed correctly there.  The event being handled here is the deletion of the first replicaset

Reply:
Triggers get merged by the work queue, so I don't know that that's guaranteed to be the case.

Reply:
still not convinced this early exit is a good idea.

Reply:
I don't think a live lookup is the correct fix here. It would make more sense to me to track the replicaset uid alongside the associated expectations, and if it mismatches, delete the expectations on sync